Arduino/MiP_ProMini_Pack_Library/src/mip.cpp

```diff
@@ -1937,6 +1951,242 @@ int8_t MiP::rawGetUserData(uint8_t address, uint8_t& userData)
 
 
 
+void MiP::enableMiPDetectionMode(uint8_t id, uint8_t txPower)
+{
+    m_irId = id;
+
+    rawSetMiPDetectionMode(id, txPower);
+}
+
+void MiP::disableMiPDetectionMode()
+{
+    m_irId = MIP_IR_DETECTION_MODE_DISABLE;
+
+    // According to WowWee documentation, TX power must be between 1 and 120 even when disabling.
+    rawSetMiPDetectionMode(MIP_IR_DETECTION_MODE_DISABLE, 0x01);
+}
+
+bool MiP::isMiPDetectionModeEnabled()
+{
+    return m_irId > MIP_IR_DETECTION_MODE_DISABLE ? true : false;
+}
+
+MiPDetectedMiP MiP::readDetectedMiP()
+{
+    // Fetch bytes from the Serial receive buffer and process any event data found within.
+    processAllResponseData();
+
+    MiPDetectedMiP detectedMiPEvent;
+    detectedMiPEvent.detected = false;
+    if(!m_detectedMiPEvents.pop(detectedMiPEvent))
+    {
+        m_lastError = MIP_ERROR_NO_EVENT;
+        return detectedMiPEvent;
+    }
+    m_lastError = MIP_ERROR_NONE;
+    return detectedMiPEvent;
+}
+
+uint8_t MiP::availableDetectedMiPEvents()
+{
+    // Fetch bytes from the Serial receive buffer and process any event data found within.
+    processAllResponseData();
+    m_lastError = MIP_ERROR_NONE;
+    return m_detectedMiPEvents.available();
+}
+
+// This internal protected method sends the set detection mode command with minimal error
+// handling. The error recovery happens at a higher level of the driver.
+void MiP::rawSetMiPDetectionMode(uint8_t id, uint8_t txPower)
+{
+    uint8_t command[1+2];
+
+    MIP_ASSERT( MIP_IR_DETECTION_MODE_DISABLE <= id && id <= 0xFF );
```

Comment:
You can probably remove this assert since it just asserts that id falls between 0x00 and 0xFF, the only allowed values for a uint8_t anyway.

Reply:
Good point.  Will remove later today.